src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -337,8 +337,20 @@ impl<'a> Resolver<'a> {
 
             // These items live in both the type and value namespaces.
             ItemKind::Struct(ref struct_def, _) => {
+                if self.session.features.borrow().non_exhaustive {
```

Comment:
Unnecessary condition again, you don't have to check this all the time, one feature check in `feature_gate.rs` is enough.

Reply:
Fixed. Wasn't aware that I didn't need it everywhere - I misinterpreted the instructions on the [Rust forge](https://forge.rust-lang.org/feature-guide.html).